### DISCUSSION TITLE
Dynamically adding new nodes with new positions to batch

### DISCUSSION THREAD

**Discussion by j-merz** [pyg-team/pytorch_geometric#2799] [Q&A]
Hi,

I have a recurrent neural network that predicts new node positions. I would like to simply add these to the respective graphs in my batch. Is there an easy way to do it that keeps the Batch datastructure intact?

The only way I could find so far, was updating the underlying data and having a dataloader generate the batch again for me. There should be an easier way, right?

Thanks a lot,
J

**Comment by rusty1s** [comment]
Can you show me an example? Do you want to add new nodes or just change attributes of your graph? You can, of course, also add such data to your already created batch, e.g., by just replacing those features in `batch.pos` or via concatenating. In case of adding new edges, however, you need to keep track of correctly increasing edge indices according to the ordering of your mini-batch.
